pkg/coremodel/pluginmeta/coremodel.cue

```diff
@@ -11,18 +15,19 @@ seqs: [
 				// Unique name of the plugin. If the plugin is published on
 				// grafana.com, then the plugin id has to follow the naming
 				// conventions.
-				id: =~"^[0-9a-z]+\\-([0-9a-z]+\\-)?(app|panel|datasource)$"
+				id: =~"^[0-9a-z]+\\-([0-9a-z]+\\-)?(\(strings.Join([for t in _types {t}], "|")))$"
+
+				// types indicates the set of all plugin types. This private field solely
+				// exists so that the plugin type can be interpolated in other fields
```

Comment:
nitting on myself

```suggestion
				// The set of all plugin types. This hidden field exists solely
				// so that the set can be string-interpolated into other fields.
```